homeassistant/components/medcom_ble/config_flow.py

```diff
@@ -0,0 +1,164 @@
+"""Config flow for Medcom BlE integration."""
+
+from __future__ import annotations
+
+import dataclasses
+import logging
+from typing import Any
+
+from bleak import BleakError
+from bluetooth_data_tools import human_readable_name
+from medcom_ble import MedcomBleDevice, MedcomBleDeviceData
+import voluptuous as vol
+
+from homeassistant.components import bluetooth
+from homeassistant.components.bluetooth import (
+    BluetoothServiceInfo,
+    async_discovered_service_info,
+)
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class Discovery:
+    """A discovered bluetooth device."""
+
+    name: str
+    discovery_info: BluetoothServiceInfo
+    device: MedcomBleDevice
+
+
+def get_name(device: MedcomBleDevice) -> str:
+    """Generate name with model and identifier for device."""
+
+    name = device.friendly_name()
+    if identifier := device.identifier:
+        name += f" ({identifier})"
+    return name
+
+
+class InspectorBLEConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Medcom BLE radiation monitors."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._discovery_info: BluetoothServiceInfo | None = None
+        self._discovered_device: Discovery | None = None
+        self._discovered_devices: dict[str, Discovery] = {}
+
+    async def _get_device_data(
+        self, discovery_info: BluetoothServiceInfo
+    ) -> MedcomBleDevice:
+        ble_device = bluetooth.async_ble_device_from_address(
+            self.hass, discovery_info.address
+        )
+        if ble_device is None:
+            _LOGGER.debug("no ble_device in _get_device_data")
+            raise BleakError("Cannot find BLE device")
+
+        inspector = MedcomBleDeviceData(_LOGGER)
+
+        try:
+            data = await inspector.update_device(ble_device)
+        except BleakError as err:
```

Comment:
Since you are already catching these below, these can be removed

Reply:
I'm not sure I understand - I'm catching a more specific error first to provide better context if the logs if it occurs, then a generic exception to make the integration more robust in case of unforeseen issues with the underlying library, why is this not good?

Reply:
We don't want error logging in config flows except for a `_LOGGER.exception` for the broad exception catch for two reasons:

1. The user is not expected to be tailing the logs while setting up the integration and won't see the reason the exception happened.  The UI should give the user enough feedback to know what to do when they encounter an exception instead.
2. We don't want to spam the logs for exceptions we know have been handled as it makes analyzing log files more difficult, wastes cpu time, and fills the logs ui with garbage.